.github/workflows/test-release.yaml

```diff
@@ -17,6 +17,8 @@ jobs:
       # Fetch the latest commit
       - name: Checkout
         uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
**Question:**

I see [here](https://github.com/actions/checkout) that `fetch-depth: 0` 
>indicates all history for all branches and tags

Why is this needed? 
Is it to always run `CommitLint` in all branches every time? 
(Sorry for the noob GitHub Workflows question 🙈 )

Reply:
It's a good question.

Without this github does a very limited checkout that only fetched the branch that triggered the workflow (i.e. the PR branch).

But below we pass `-f origin/main` to only run on the commit messages starting "from" `origin/main`.  For this to work, git/commitlint needs to know which commits exist on `origin/main` which is why we need to fetch all branches.